File: leet_MATH_2033_uni_val.cc
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x)
    {
        vector<int(*)(vector<vector<int>>&, int)> Solutions {
            median_and_flattened,
        };
        return Solutions[0](grid, x);
    }

    static int median_and_flattened(vector<vector<int>> &g, int x)
    {
        vector<int> F;
        for (auto r : g){
            for (int n : r){
                F.push_back(n);
            }
        }
        int N = F.size();
        sort(F.begin(), F.end());
        int median = F[N /2 ];
        int res = 0;
        for (int n : F) {
            int diff = abs(n - median);
            if (diff % x)
                return -1;
            res += diff / x;
        }
        return res;
    }
};

File: leet_MATH_2033_uni_val.go
func minOperations(grid [][]int, x int) int {
    Solutions := []func([][]int,int) int {
        median_and_flattened,
    }
    return Solutions[0](grid, x)
}

func median_and_flattened(g [][]int, x int) int {
    F := []int{}
    for _,r := range g {
        for _,n := range r {
            F = append(F,n)
        }
    }
    sort.Ints(F)
    N := len(F)
    median := F[N / 2]
    res := 0
    for _,n := range F {
        diff := n - median
        if diff < 0 { diff = -diff }
        if diff % x != 0 {
            return -1
        }
        res += diff / x
    }
    return res
}
